content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -218,6 +289,27 @@ this variable to checkout the appropriate version of the library.
 
 image::pipeline/global-pipeline-library-legacy-scm.png["Configuring a 'Legacy SCM' for a Pipeline Library", role=center]
 
+==== Dynamic retrieval
```

Comment:
Is the Dynamic Retrieval something which is _only_ going to be supported by the `library` step? If so, I suggest grouping it with the documentation sections above describing the `library` step